codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java

```diff
@@ -204,7 +205,9 @@ private State readHeaders(ByteBuf buffer, StompHeaders headers) {
                 } else {
                     int globalIndex = indexOf(buffer, buffer.readerIndex(),
                             buffer.writerIndex(), StompConstants.NUL);
-                    if (globalIndex != -1) {
+                    if (globalIndex == -1) {
+                        throw Signal.valueOf(ReplayingDecoder.class, "REPLAY");
```

Comment:
If a NUL byte isn't found in the remainder of the buffer it raises a REPLAY signal to continue searching in the next chunk of data.